server/remark-code-snippet.ts

```diff
@@ -42,43 +41,91 @@ const isCode =
   (node: MdxastNode): node is MdastCode =>
     node.type === "code" && langs.includes(node.lang);
 
-const getCommandNode = (content: string, prefix = "$"): MdxJsxFlowElement => ({
-  type: "mdxJsxFlowElement",
-  name: "Command",
-  attributes: [],
-  children: [
-    {
-      type: "mdxJsxFlowElement",
-      name: "CommandLine",
-      attributes: [
-        {
-          type: "mdxJsxAttribute",
-          name: "data-content",
-          value: `${prefix} `,
-        },
-      ],
-      children: [
-        {
-          type: "text",
-          value: content,
-        },
-      ],
-    },
-  ],
+const getTextChildren = (contentValue: string): MdxastNode => ({
+  type: "text",
+  value: contentValue,
 });
 
-const getLineNode = (content: string, attributes = []): MdxJsxFlowElement => ({
+const getVariableNode = (
+  valueName: string,
+  isGlobal: boolean
+): MdxJsxFlowElement => ({
   type: "mdxJsxFlowElement",
-  name: "CommandLine",
-  attributes,
-  children: [
+  name: "Var",
+  attributes: [
+    { type: "mdxJsxAttribute", name: "name", value: valueName },
     {
-      type: "text",
-      value: content,
+      type: "mdxJsxAttribute",
+      name: "isGlobal",
+      value: createMdxJsxAttributeValueExpression(`${isGlobal}`),
     },
   ],
+  children: [],
 });
 
+const getChildrenNode = (content: string): MdxastNode[] => {
+  const hasVariable = content?.includes("<Var");
+  const nodeChildren: MdxastNode[] = [];
+
+  if (hasVariable) {
+    const contentVars = content.match(/(?:\<Var .+?\/\>)/gm);
+    const firstPartLine = content.split("<Var")[0];
+    nodeChildren.push(getTextChildren(firstPartLine));
+    const newContent = content.replace("isGlobal", "");
+
+    for (let i = 0; i < contentVars.length; i++) {
+      let nextPartLine = newContent.split("/>")[i + 1];
+      if (nextPartLine?.includes("<Var")) {
+        nextPartLine = nextPartLine.split("<Var")[0];
+      }
+
+      const varName = contentVars[i].match(/name="(.*?)"/)[1];
+      const isGlobal = contentVars[i].includes("isGlobal");
+      nodeChildren.push(getVariableNode(varName, isGlobal));
+      nodeChildren.push(getTextChildren(nextPartLine));
+    }
```

Comment:
Why not `contentVars.forEach((item, i) =>{
...
})`? If there is reason to use a for loop, then perhaps use `for(let item in contentVars)`

Reply:
`for` and `forEach` loops are similar. But a `for` loop will gives you more control and doesn't require a new function creation. I prefer to leave `for` if you don't against